src/gui/sdl_mapper.cpp

```diff
@@ -1754,41 +1764,135 @@ static void change_action_text(const char* text,Bit8u col) {
 	bind_but.action->SetColor(col);
 }
 
+static std::string humanize_mod_name(const CBindList &binds, const std::string &fallback)
+{
+	auto trim_pfx = [](const std::string &pfx, const std::string &input) {
+		if (starts_with(pfx, input))
+			return input.c_str() + pfx.size();
+		else
+			return input.c_str();
+	};
+
+	const auto binds_num = binds.size();
+
+	// We have a single bind, just use it
+	if (binds_num == 1)
+		return binds.front()->GetBindName();
+
+	// Avoid prefix, e.g. "Left Alt" and "Right Alt" -> "Alt"
+	if (binds_num == 2) {
+		const std::string new_pfx = (fallback.empty() ? "" : fallback + ": ");
+		const std::string name_1 = binds.front()->GetBindName();
+		const std::string name_2 = binds.back()->GetBindName();
+		std::string trimmed_1 = trim_pfx("Right ", name_1);
+		std::string trimmed_2 = trim_pfx("Left ", name_2);
+		if (trimmed_1 == trimmed_2)
+			return new_pfx + trimmed_1;
+		trimmed_1 = trim_pfx("Left ", name_1);
+		trimmed_2 = trim_pfx("Right ", name_2);
+		if (trimmed_1 == trimmed_2)
+			return new_pfx + trimmed_1;
```

Comment:
This is a bit of a maze.. name_1, name_2, trimmed_1, trimmed_2, and then the two evaluations are the same`if (trimmed_1 == trimmed_2` but with a different assignment. I guess if neither of those pass we return the fallback.

I'm not sure if we can use more descriptive names for the _1 and _2; I can see that one is the front and the other is the back, and similarly right and left.

Reply:
These are literally the first and second key names :) (there are only 2 in this case).

Maybe I can make this algorithm cleaner by using regex instead...

Reply:
Ahh, in that case maybe just a bit more commenting of what the code is trying to accomplish (just this specific block). If that's what the names are, then we can't go much further.

(Unless regex would really be the more elegant and general solution?)

Reply:
Simplified this whole block, should be much more readable now :)